### PR TITLE
Multi language

### PR DESCRIPTION
**Pull request by LeeCampbell**
JAVA!
This PR brings in a Java port using JavaSpark as the Web Host implementation.

Pretty similar design to the C# code, with some concessions to Java being a different language with different features.

Include CircleCI Build Server config which compile and test the domain.

### CODE REVIEW

File: java/src/main/java/com/leecampbell/cod/application/SparkApplication.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        CreateLoanModel requestModel = gson.fromJson(requestBody, CreateLoanModel.class);
+        CreateLoanCommand command = requestModel.toCommand();
+
+        Repository repository = new FakeRepository();
+        CreateLoanCommandHandler handler = new CreateLoanCommandHandler(repository);
+        Receipt receipt = handler.handle(command);
+
+        LoanCreatedModel responseModel = new LoanCreatedModel();
+        responseModel.loanId = receipt.aggregateId();
+        return gson.toJson(responseModel);
```
Comment (vgw-rhysc) [inline java/src/main/java/com/leecampbell/cod/application/SparkApplication.java]: can use json response transformers here if you want
http://sparkjava.com/documentation#response-transformer

File: java/src/main/java/com/leecampbell/cod/domain/model/AggregateRoot.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+import com.leecampbell.cod.domain.contracts.DomainEvent;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.*;
+
+public abstract class AggregateRoot {
+
+    private final UUID id;
+    private final List<DomainEvent> uncommittedEvents = new ArrayList<DomainEvent>();
+    //private static Map<String, Method> mutatorMethods = new HashMap<String, Method>();
```
Comment (vgw-rhysc) [inline java/src/main/java/com/leecampbell/cod/domain/model/AggregateRoot.java]: commented code

File: java/src/main/java/com/leecampbell/cod/domain/model/AggregateRoot.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    public void ApplyEvent(DomainEvent payload) {
+        Class<? extends AggregateRoot> rootType = this.getClass();
+        Class<? extends DomainEvent> eventType = payload.getClass();
+        try {
+            Method handler = getMethod(payload);
+            handler.setAccessible(true);
+			handler.invoke(this, payload);
+		} catch (IllegalAccessException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
```
Comment (vgw-rhysc) [inline java/src/main/java/com/leecampbell/cod/domain/model/AggregateRoot.java]: why not use a logger?

File: java/src/main/java/com/leecampbell/cod/domain/model/AggregateRoot.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+        Method method = null;
+
+        try {
+
+            // assume protected or private...
+
+            method = rootType.getDeclaredMethod("handle", eventType);
+
+        } catch (Exception e) {
```
Comment (vgw-rhysc) [inline java/src/main/java/com/leecampbell/cod/domain/model/AggregateRoot.java]: is it a more specific exception? ie `NoSuchMethodException`, `SecurityException`

File: java/src/main/java/com/leecampbell/cod/domain/model/Loan.java [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+            throw new UnsupportedOperationException("Transaction amount must be positive.");
+
+        AddEvent(new PaymentTakenEvent(UUID.randomUUID().toString(), command.transactionDate(),
+                command.transactionAmount()));
+        if (balance.compareTo(BigDecimal.ZERO) >= 0)
+            AddEvent(new LoanSettledEvent(command.transactionDate()));
+        if (balance.compareTo(BigDecimal.ZERO) > 0)
+            AddEvent(new LoanOverPaidEvent(command.transactionDate(), balance));
+    }
+    //public void ReverseDisbursement(ReverseDisbursement command) {
```
Comment (vgw-rhysc) [inline java/src/main/java/com/leecampbell/cod/domain/model/Loan.java]: Commented code

File: java/src/main/java/com/leecampbell/cod/domain/model/AggregateRoot.java (hunk 3 above)
Comment (LeeCampbell) [reply java/src/main/java/com/leecampbell/cod/domain/model/AggregateRoot.java]: `Auto-generated catch block`!

**Pull request merged**